opal/mca/btl/openib/btl_openib_component.c

```diff
@@ -2635,23 +2635,10 @@ btl_openib_component_init(int *num_btl_modules,
         goto no_btls;
     }
 
-    /* If we want fork support, try to enable it */
-#ifdef HAVE_IBV_FORK_INIT
-    if (0 != mca_btl_openib_component.want_fork_support) {
-        if (0 != ibv_fork_init()) {
-            /* If the want_fork_support MCA parameter is >0, then the
-               user was specifically asking for fork support and we
-               couldn't provide it.  So print an error and deactivate
-               this BTL. */
-            if (mca_btl_openib_component.want_fork_support > 0) {
-                opal_show_help("help-mpi-btl-openib.txt",
-                               "ibv_fork_init fail", true,
-                               opal_process_info.nodename);
-                goto no_btls;
-            }
-        }
+    /* If fork support is requested, try to enable it */
```

Comment:
I hear what you're saying.

Let me offer an alternative.

Have a look at opal/mca/common/verbs/common_verbs_mca.c.  It registers a "real" MCA var, and then it provides an API for upper layers to call to register a synonym for that "real" MCA var.  In this way, you can preserve the abstraction boundaries.

The reason I'm asking for this synonym is twofold:

1. We can't just kill an MCA var that users may be using.  You noted in another comment that no one is likely using the old param because it didn't work.  But I think there actually was a sizeable amount of time where this param *did* work (i.e,. before we introduced the UD OOB).  So there may be people out there using it.
1. The behavior of ibv_fork_init() is global to the entire process -- it's not specific to an individual component.  Hence, asking for fork support with the openib component is really also asking for fork support in all other verbs-enabled components.